### PR TITLE
SQL JOINs

### PR DESCRIPTION
**Pull request by mihaiberq**
- 👶💪JOINs
  - [x] INNER JOIN
  - [x] LEFT / RIGHT JOIN
  - [x] FULL JOIN
  - [x] Multiple JOINs

### CODE REVIEW

**Comment by enki-hq** [comment]
- [ ] https://enkipro.com/preview-insight/sagelabs/content/SQL/DQL/joins/full-join.md?ref=0c3a7fb8bb57923979f6d700c68aec5c8f3b8d4f
- [ ] https://enkipro.com/preview-insight/sagelabs/content/SQL/DQL/joins/inner-join.md?ref=0c3a7fb8bb57923979f6d700c68aec5c8f3b8d4f
- [ ] https://enkipro.com/preview-insight/sagelabs/content/SQL/DQL/joins/left-and-right-joins.md?ref=0c3a7fb8bb57923979f6d700c68aec5c8f3b8d4f
- [ ] https://enkipro.com/preview-insight/sagelabs/content/SQL/DQL/joins/multiple-joins.md?ref=0c3a7fb8bb57923979f6d700c68aec5c8f3b8d4f

**Comment by mihaiberq** [comment]
@lizTheDeveloper Sorry about that, changed master to `sql` branch and after I opened the PR and forgot to double check. Moved everything to this one.

File: SQL/DQL/joins/full-join.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+1   |        pound      |   1    |  normal
+...
+1234| coolest-move-ever |  NULL  |   NULL
+NULL|        NULL       |   19   |   wood
+```
+
+---
+## Practice
+
+The `FULL OUTER JOIN` operations is equivalent to
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/full-join.md]: have users do a full outer join with fill in the gaps

File: SQL/DQL/joins/full-join.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+---
+## Revision
+
+The correct syntax for fully joining two tables is:
+```SQL
+SELECT *
+FROM pokemon_type
+... type ON
+pokemon_type.type_id = type.id
+```
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/full-join.md]: do a mulit-part gap fill, add the ON part into the ??? slots

File: SQL/DQL/joins/inner-join.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+An *INNER JOIN* is a type of join that only returns rows for which the joined field (`id` for `ability` and `ability_id` for `ability_effect_text`) are common: if there were some abilities without an effect text or some effects not describing an ability, they wouldn't be included in query result.
+
+There were a total of `191` rows returned by the query versus `251` abilities and `191` text effects.
+
+---
+## Footnotes
+
+[1: Ability]
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/inner-join.md]: 💯  for accurate footnotes

File: SQL/DQL/joins/inner-join.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+* intersection
+* union
+* difference
+* cartesian product
+
+---
+## Revision
+
+Fill in the blanks such that the following snippet contains a valid JOIN operations:
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/inner-join.md]: switch the practice and revision question

File: SQL/DQL/joins/multiple-joins.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+   - workout
+
+---
+## Content
+
+Sometimes, a single join might not be enough to get the desired result. In `many-to-many` relationships, when an intermediate table is used to avoid data duplication, such practice is common.
+
+In the Pokémon database you can find multiple intermediate tables, usually named `table1_table2`: for the `pokemon` and `type` tables, the intermediate one is named `pokemon_type`. Same goes for `pokemon`, `move` and `pokemon_move`.
+
+A single JOIN is simple to achieve: take a table, use the magic word (`type JOIN`), specify the second table and choose a joining field. However, you can also join multiple tables together.
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/multiple-joins.md]: you don't need this sentence

File: SQL/DQL/joins/multiple-joins.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+name        | name
+
+bulbasaur   | grass
+bulbasaur   | poison
+```
+The same result can be achieved by using subqueries.
+
+---
+## Practice
+
```
Comment (lizTheDeveloper) [inline SQL/DQL/joins/multiple-joins.md]: make this a fill-the-gaps question

**Pull request merged**